src/Core/Resolvers/MultipleCreateOrderHelper.cs

```diff
@@ -61,6 +64,26 @@ public class MultipleCreateOrderHelper
                     subStatusCode: DataApiBuilderException.SubStatusCodes.EntityNotFound);
             }
 
+            DatabaseTable sourceDbTable = (DatabaseTable)sourceDbObject;
+            DatabaseTable targetDbTable = (DatabaseTable)targetDbObject;
+            if (sourceDbTable.Equals(targetDbTable))
+            {
+                //  DAB does not yet support multiple-create for self referencing relationships where both the source and
+                //  target entities are backed by same database table.
+                throw new DataApiBuilderException(
+                message: $"Multiple-create for relationship: {relationshipName} at level: {nestingLevel} is not supported because " +
```

Comment:
Are queries supported when there is a relationship between entities backed by same tables? 

And the PR title only says mutation is not allowed. We should be specific in the title to qualify that "multiple" mutation isnt allowed.

Reply:
@seantleonard, could you help answer the first question? I'm assuming we will have the support atleast for self-referencing relationships, right?

Reply:
for question 1: 
- #2138 handles a single entity which defines "self-joining" relationships
- TBD on two separate entities which define relationships to each other. TBD because I think there is an open issue and then I need to test within pr 2138 if the scenario works. 
Shouldn't block this pr though.